Automattic/newspack-blocks
# Fix/modal checkout tweaks

### All Submissions:






### Changes proposed in this Pull Request:



Tweaks to the WC modal checkout introduced in #1355 – the JS added for the modal checkout should not be loaded if the transaction won't be processed by WC. 

### How to test the changes in this Pull Request:

1. On `master`, set Stripe as the Reader Revenue platform
2. Load a page with the Donate block and inspect the `/wp-content/plugins/newspack-blocks/dist/donate/view.js` file – it will contain modal-related JS (which will not be used)
3. Inspect page content, observe a `div.newspack-blocks-donate-checkout-modal` element is rendered in the footer (which will not be used)
4. Switch to this branch, rebuild (restart webpack!), observe not extra JS content and no extra markup in the footer
5. Switch RR platform to Newspack, ensure the modal checkout flow still works

### Other information:
